src/ILCompiler.Compiler/src/Compiler/TypeExtensions.cs

```diff
@@ -201,7 +201,7 @@ public static bool IsArrayTypeWithoutGenericInterfaces(this TypeDesc type)
         /// </summary>
         public static bool RequiresAlign8(this TypeDesc type)
         {
-            if (type.Context.Target.Architecture != TargetArchitecture.ARM)
+            if (type.Context.Target.Architecture != TargetArchitecture.ARM && type.Context.Target.Architecture != TargetArchitecture.ARMEL)
```

Comment:
Are we actually getting anything from having `TargetArchitecture.ARMEL`? Or do we expect to get something out of it in the future?

From a cursory search, it seems like we treat it the same as `TargetArchitecture.ARM` everywhere, and where we don't, it's just a latent bug that someone will hit (e.g. the occurrences under `src/System.Private.TypeLoader/`).

Reply:
That was my observation too, and I don't think we'll need it in the future. Would it be ok to remove it in this PR, or should I do it in a separate one?

Reply:
> Would it be ok to remove it in this PR, or should I do it in a separate one?

It would be nice to have the two distinct changes in two separate commits, but it can stay in a single pull request.

Reply:
Ok, I removed `TargetArchitecure.ARMEL`